Lesson1/app/src/main/java/com/gmail/l2t45s7e9/lesson1/ContactService.java

```diff
@@ -1,21 +1,73 @@
 import android.app.Service;
+import android.content.ContentResolver;
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
 import android.os.Binder;
+import android.os.Build;
 import android.os.IBinder;
+import android.provider.ContactsContract;
+import android.widget.ArrayAdapter;
+import android.widget.Toast;
+
+import androidx.annotation.RequiresApi;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+
 import java.lang.ref.WeakReference;
+import java.util.ArrayList;
 import java.util.Calendar;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.GregorianCalendar;
 
 public class ContactService extends Service {
 
     interface Informator{
         ContactService getService();
     }
-    private static People first = new People("Иван", "+7919 151 6321", "999", "1@gmail.com", "3@gmail.com", "Описание",new GregorianCalendar(1985, Calendar.JUNE,6));
-    private static People second = new People("Дима", "+7912 112 4577", "888", "2@gmail.com", "4@gmail.com", "Описание",new GregorianCalendar(1985, Calendar.JUNE,18));
-    static People[] people = {first, second};
+    ArrayList<People> contacts = new ArrayList<People>();
+
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        ContentResolver contentResolver = getContentResolver();
+        Cursor cursor = contentResolver.query(ContactsContract.Data.CONTENT_URI,
+                new String[] {ContactsContract.Contacts.HAS_PHONE_NUMBER,
+                        ContactsContract.Data.DISPLAY_NAME,
+                        ContactsContract.CommonDataKinds.Email.ADDRESS,
+                        ContactsContract.CommonDataKinds.Phone.NUMBER},
+                null,
+                null,
+                ContactsContract.Data.DISPLAY_NAME);
+
+        if(cursor!=null){
+
+            while (cursor.moveToNext()){
+                String hasNumber = cursor.getString(
+                        cursor.getColumnIndex(ContactsContract.Contacts.HAS_PHONE_NUMBER));
+
+                String email = cursor.getString(
+                        cursor.getColumnIndex(ContactsContract.CommonDataKinds.Email.ADDRESS));
+                String name = cursor.getString(
+                        cursor.getColumnIndex(ContactsContract.Contacts.DISPLAY_NAME));
+                String number = cursor.getString(
+                        cursor.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
+                if(hasNumber.equalsIgnoreCase("1") && !name.equalsIgnoreCase(number)){
+                        People people = new People(name,number,number,email ,email,getString(R.string.description) , new GregorianCalendar(1985, Calendar.JUNE,18) );
+                        contacts.add(people);
+                    }
+                }
+            }
+            cursor.close();
```

Comment:
https://github.com/goblinr/AndroidLessonsCommonMistakes/blob/master/%D0%97%D0%B0%D0%B3%D1%80%D1%83%D0%B6%D0%B0%D1%82%D1%8C_%D0%BA%D0%BE%D0%BD%D1%82%D0%B0%D0%BA%D1%82%D1%8B_%D0%B2_%D1%81%D0%B5%D1%80%D0%B2%D0%B8%D1%81%D0%B5_%D0%B8%D0%B7_%D0%BF%D0%BE%D1%81%D1%82%D0%B0%D0%B2%D1%89%D0%B8%D0%BA%D0%B0_%D0%BA%D0%BE%D0%BD%D1%82%D0%B0%D0%BA%D1%82%D0%BE%D0%B2.md#%D0%BD%D0%B5%D0%B2%D0%B5%D1%80%D0%BD%D0%BE-%D0%B7%D0%B0%D0%BA%D1%80%D1%8B%D0%B2%D0%B0%D1%8E%D1%82-%D0%BA%D1%83%D1%80%D1%81%D0%BE%D1%80